src/coreclr/jit/lowerxarch.cpp

```diff
@@ -7511,6 +7600,154 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
     }
 }
 
+//----------------------------------------------------------------------------------------------
+// MakeHWIntrInsicSrcContained: Unfold the eligible constant vector when embedded broadcast is
+//                              available.
+//
+//  Arguments:
+//     parentNode - The hardware intrinsic node
+//     childNode  - The operand node to try contain
+//
+void Lowering::MakeHWIntrinsicSrcContained(GenTreeHWIntrinsic* parentNode, GenTree* childNode)
+{
+    assert(childNode->OperIs(GT_CNS_VEC));
+    GenTreeVecCon* vecCon = childNode->AsVecCon();
+    if (vecCon->IsAllBitsSet() || vecCon->IsZero())
+    {
+        // do not enable embedded broadcast for all 1/0 vectors.
+        MakeSrcContained(parentNode, childNode);
```

Comment:
I don't think we want to contain `zero`/`allbitsset` here. Those will need a register and need to have codegen generate `pcmpeqd` or `xorps`

Reply:
We probably want to assert we don't reach here for them and that they aren't marked as containable (I think we might already be filtering that for the most part).